File: JavaScript/ComplexNumberMultiplication/ComplexNumberMultiplication.js
/**
 * @param {string} a
 * @param {string} b
 * @return {string}
 */
 // (102ms)
var complexNumberMultiply = function(a, b) {
    var numA = a.slice(0,-1).split('+').map(Number);
    var numB = b.slice(0,-1).split('+').map(Number);
    var real = numA[0]*numB[0] - numA[1]*numB[1];
    var virtual = numA[0]*numB[1] + numA[1]*numB[0];
    return `${real}+${virtual}i`;
};

File: Python/ComplexNumberMultiplication/ComplexNumberMultiplication.py
class Solution(object):
	# (38ms)
    def complexNumberMultiply(self, a, b):
        """
        :type a: str
        :type b: str
        :rtype: str
        """
        numA, numB = map(int, a[:-1].split('+')), map(int, b[:-1].split('+'))
        return '%d+%di' % (numA[0]*numB[0] - numA[1]*numB[1], numA[0]*numB[1] + numA[1]*numB[0])